docs/src/fock.md

```diff
@@ -0,0 +1,147 @@
+# Fock States
```

Comment:
This page mentions in a couple of places things that are not Fock states. Maybe we can clarify this somewhere in the introductory paragraphs and change the title. Something like "Quantum Harmonic Oscillators" or "Bosonic Systems". And a description of how "we call the basis a `FockBasis` but we can symbolically represent arbitrary bosonic states in it".

Reply:
Maybe it would be good in the future to have boson and fermion bases in QuantumInterface?